flocker/control/test/test_httpapi.py

```diff
@@ -242,6 +242,76 @@ def test_container_name_collision_different_node(self):
         """
         return self._container_name_collision_test(self.NODE_A, self.NODE_B)
 
+    def test_create_container_with_environment(self):
+        """
+        A valid API request to create a container including environment
+        variables results in an updated configuration.
+        """
+        saving = self.persistence_service.save(Deployment(
```

Comment:
I'd have thought that behaviour would be tested, once, separately, not for every variation of the create container parameters.

But if you want to keep it this way, perhaps it could be made clearer by assigning to a variable called `original_configuration` and then using the `Deployment.update_node` method to derive the expected `updated_configuration`. Up to you. 

But at least make the test docstring a little clearer.